lib/generate-suggestions.ts

```diff
@@ -22,88 +23,91 @@ interface Occurrence {
 
 async function generateSuggestions(flags: { directory?: string }) {
   const components = await fetchComponents();
-  const results: { [apiId: string]: Result } = {};
+  const directory = flags.directory || ".";
 
-  for (const [compApiId, component] of Object.entries(components)) {
-    if (!results[compApiId]) {
-      results[compApiId] = { apiId: compApiId, ...component, occurrences: {} };
-    }
-
-    const directory = flags.directory || ".";
-    const result = await findTextInJSXFiles(directory, component);
-    results[compApiId].occurrences = result;
-
-    // Remove if there the length is zero
-    if (Object.keys(results[compApiId].occurrences).length === 0) {
-      delete results[compApiId];
-    }
-  }
+  const results: { [apiId: string]: Result } = await findComponentsInJSXFiles(
+    directory,
+    components
+  );
 
   // Display results to user
   console.log(JSON.stringify(results, null, 2));
 }
 
-async function findTextInJSXFiles(
+async function findComponentsInJSXFiles(
   path: string,
-  component: FetchComponentResponseComponent
-) {
-  const result: Result["occurrences"] = {};
+  components: FetchComponentResponse
+): Promise<{ [apiId: string]: Result }> {
+  const result: { [apiId: string]: Result } = {};
   const files = glob.sync(`${path}/**/*.+(jsx|tsx)`, {
     ignore: "**/node_modules/**",
   });
 
   const promises: Promise<any>[] = [];
 
   for (const file of files) {
-    result[file] = [];
-
     promises.push(
       fs.readFile(file, "utf-8").then((code) => {
         const ast = parse(code, {
           sourceType: "module",
           plugins: ["jsx", "typescript"],
         });
 
-        const occurrences: Occurrence[] = [];
-
         traverse(ast, {
           JSXText(path) {
-            if (path.node.value.includes(component.text)) {
-              const escapedText = component.text.replace(
-                /[.*+?^${}()|[\]\\]/g,
-                "\\$&"
-              );
-              const regex = new RegExp(escapedText, "g");
-              let match;
-              while ((match = regex.exec(path.node.value)) !== null) {
-                const lines = path.node.value.slice(0, match.index).split("\n");
-
-                if (!path.node.loc) {
-                  continue;
-                }
-
-                const lineNumber = path.node.loc.start.line + lines.length - 1;
+            for (const [compApiId, component] of Object.entries(components)) {
```

Comment:
As discussed on call, we can load components into a map and then get them in `O(1)` time inside of the tree traversal which should speed things up even more!
```jsx
const matchingComponents = componentMap.get(path.node.value);
if (!matchingComponents) return;